tests/e2e/test_rosacli_upgrade.go

```diff
@@ -305,6 +305,120 @@ var _ = Describe("Cluster Upgrade testing",
 		})
 	})
 
+var _ = Describe("Describe/List rosa upgrade",
+	labels.Feature.Cluster, func() {
+		defer GinkgoRecover()
+		var (
+			rosaClient     *rosacli.Client
+			clusterService rosacli.ClusterService
+			upgradeService rosacli.UpgradeService
+			clusterID      string
+			profile        *profilehandler.Profile
+		)
+
+		BeforeEach(func() {
+			By("Get the cluster")
+			clusterID = config.GetClusterID()
+			Expect(clusterID).ToNot(Equal(""), "ClusterID is required. Please export CLUSTER_ID")
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			clusterService = rosaClient.Cluster
+			upgradeService = rosaClient.Upgrade
+
+			By("Load the profile")
+			profile = profilehandler.LoadProfileYamlFileByENV()
+		})
+
+		AfterEach(func() {
+			By("Delete cluster upgrade")
+			output, err := clusterService.DeleteUpgrade("-c", clusterID, "-y")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output.String()).To(ContainSubstring("Successfully canceled scheduled upgrade on cluster '%s'", clusterID))
+
+			By("Clean remaining resources")
+			err = rosaClient.CleanResources(clusterID)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("to list/describe rosa upgrade via ROSA CLI - [id:57094]",
+			labels.High, labels.Runtime.Day2,
+			func() {
+				By("Check the help message of 'rosa describe upgrade -h'")
+				output, err := clusterService.DescribeUpgrade(clusterID, "-h")
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("rosa describe upgrade [flags]"))
+				Expect(output.String()).To(ContainSubstring("-c, --cluster"))
+				Expect(output.String()).To(ContainSubstring("--machinepool"))
+				Expect(output.String()).To(ContainSubstring("-y, --yes"))
+
+				By("Check list upgrade for the cluster with latest version")
```

Comment:
I would put the `By` inside the `if` clause

Reply:
update according to comment.